src/databricks/sql/ae.py

```diff
@@ -225,6 +228,15 @@ def last_sync_timestamp(self) -> Optional[datetime]:
         """The timestamp of the last time self.status was synced with the server"""
         return self._last_sync_timestamp
 
+    @property
+    def is_available(self) -> bool:
```

Comment:
How long is it available for?  How does this limitation work in the CUJ of recovering from a user-space crash?

Reply:
I'm going to check with the thrift server folks to get that answer. I believe it's available for a few hours. With cloud fetch enabled it's 24 hours.

What do you think of the name `is_available`? I'd like a name that's reflective but also easy enough to type.

Reply:
Its really a tough one; I think this might be a little misleading, because isn't the answer 'Unknown' for crash recovery CUJ?

Reply:
I don't think so. Because under this situation, if the client crashes it will do so _before_ the `query_id` is returned. Which means there would be no way to pick up the execution anyway.

Reply:
Ah, so if it gets a query_id, then it must be available, unless its been a long time?  Although, again, available doesn't sound quite right because the query could still be executing.  Maybe just returned_as_direct_result?

Reply:
Good idea. I've done the rename and pushed.

Note: we could get around this by allowing users to disable `directResults` when the `TExecuteStatementReq` is emitted.